sdk/runtime/runenv.go

```diff
@@ -99,14 +100,24 @@ type RunEnv struct {
 	RunParams
 	*logger
 
-	unstructured chan *os.File
-	structured   chan *zap.Logger
+	MetricsPusher *push.Pusher
+	unstructured  chan *os.File
+	structured    chan *zap.Logger
 }
 
 // NewRunEnv constructs a runtime environment from the given runtime parameters.
 func NewRunEnv(params RunParams) *RunEnv {
 	re := &RunEnv{
-		RunParams:    params,
+		RunParams: params,
+		MetricsPusher: push.New("http://pushgateway:9091", "testground/plan").
```

Comment:
The address of the push gateway is hardcoded here.

On the k8s runner, this the pushgateway has a service.

I could imagine making this a configurable option if/when we want to publish metrics to a remote source.

Reply:
We could pass it the same way as we do for Redis. Not sure if we are going to add support for Prometheus for the rest of the runners - `local:docker`, etc.

For now I am happy to leave it hardcoded to the k8s service address. Once we have useful dashboards, we could think about adding support for the other runners if necessary.